### ISSUE TITLE
Could you provide a dataloader.py for the Toronto3D dataset?

### ISSUE THREAD

**Issue by whuhxb** [WeikaiTan/Toronto-3D#13]
Hi @WeikaiTan 
Could you provide a dataloader.py for the Toronto3D dataset? I want to see the specific data format to suit for the semantic segmentation baseline. I'm testing my algorithm on Toronto3D dataset, but can't load the data successfully. Or, could you please provide the code of RandLA-Net on Toronto3D as you shown in the table?

In addition, have you ever run the Open3D-ML framework for Toronto3D dataset? I have run RandLANet on Toronto3D dataset, but I only get half the mIoU with TF version as reported. The pytorch version always show can't pickle bugs due to the torch.multiprocessing module.

Thanks a lot in advance.

Xiaobing Han

**Comment by WeikaiTan** [comment]
I've uploaded my code for RandLA-Net at https://github.com/WeikaiTan/RandLA-Net.git 
You may try it out.

**Comment by whuhxb** [comment]
@WeikaiTan 
Hi, I have downloaded the code and run it now. But I find that the mIoU of RandLA-Net in training stage oscillates largely, for example from 40 to 4 and then to 50. It's so strange. Have you ever met this on Toronto3D dataset when running with RandLA-Net model?

**Comment by WeikaiTan** [comment]
Yes, I experienced the same thing. First, the training and validation does not use the full point cloud, just random selection. Second, the initial learning rate is pretty large, so there will be some fluctuations in the early epochs. Third, the loss function is the cross entropy loss, so the validation accuracy is more stable compared to mIoU. It'll smooth out with more epochs trained.

**Comment by whuhxb** [comment]
> Yes, I experienced the same thing. First, the training and validation does not use the full point cloud, just random selection. Second, the initial learning rate is pretty large, so there will be some fluctuations in the early epochs. Third, the loss function is the cross entropy loss, so the validation accuracy is more stable compared to mIoU. It'll smooth out with more epochs trained.

OK. Thank you for your kind reply.

**Comment by whuhxb** [comment]
@WeikaiTan 
I have run the RandLA-Net code on Toronto3D dataset, but there is no mIoU results with the test phase. Have you ever met this? How to obtain the test result?

**Comment by WeikaiTan** [comment]
There is the test evaluation parameter in the main function

**Comment by whuhxb** [comment]
> There is the test evaluation parameter in the main function

Set test_eval as True to see the mIoU, right?

**Comment by whuhxb** [comment]
> > There is the test evaluation parameter in the main function
> 
> OK. Thanks. Have tried, and it works.

I just run RandLA-Net on Toronto3D with the code you provided, the mIoU is 72.9, much lower than the result you reported in the table. Besides, how to use the RGB information for RandLA-Net training?

**Comment by WeikaiTan** [comment]
That result was reported by Hu, author of RanLANet, in his paper. I didn't get that good result either, but I didn't spend much time trying out different parameter settings. You may change the network settings in the helper_tool.py

**Comment by whuhxb** [comment]
> That result was reported by Hu, author of RanLANet, in his paper. I didn't get that good result either, but I didn't spend much time trying out different parameter settings. You may change the network settings in the helper_tool.py

I have just check Hu's RandLA-Net paper, and haven't found the result of Toronto3D dataset, only with the introduction review of various 3D datasets but without results.

**Comment by whuhxb** [comment]
@WeikaiTan 
Have you ever run Toronto3D dataset with SuperPointGraph code? Could you please share the dataloader code? Thanks a lot.

**Comment by WeikaiTan** [comment]
Hu's results can be found at https://doi.org/10.1109/TPAMI.2021.3083288. I'll test more on the configurations to see if I could get a similar result.

**Comment by whuhxb** [comment]
> Hu's results can be found at https://doi.org/10.1109/TPAMI.2021.3083288. I'll test more on the configurations to see if I could get a similar result.

OK. I'm trying now, and waiting for your testing results. Thanks.
